locksmithctl/daemon.go

```diff
@@ -145,6 +193,10 @@ func (r rebooter) lockAndReboot(lck *lock.Lock) {
 			dlog.Warningf("Failed to acquire lock: %v. Retrying in %v.", err, interval)
 			time.Sleep(interval)
 
+			if err := getPeriodicAndSleep(); err != nil {
```

Comment:
One issue with that is that we don't reset the `interval` , it can potentially grow for a while. I think we could modify the `getPeriodicAndSleep` to return also a boolean:
* `true` in case we slept
* `false` in the other case

```suggestion
			hasSleep, err := getPeriodicAndSleep();
			if err != nil {
			    ...
			if hasSleep {
			    interval = initialInterval
			}
```

Reply:
Yes true. As we are increasing interval with `expBackoff` function it is growing but not getting reset.